htdocs/class/smarty/xoops_plugins/function.xoStats.php

```diff
@@ -0,0 +1,129 @@
+ *      $totaluser - total number of members
+ *
+ */
+
+/**
+ * @param array   $params
+ * @param \Smarty $smarty Smarty instance
+ *
+ * @return void
+ */
+function smarty_function_xoStats($params, &$smarty)
+{
+    $cache = new Cache('system');
+    $stats = $cache->cacheRead('xostats', 'regen_xoStats', 30);
+
+    foreach ($stats as $k => $v) {
+        $smarty->assign($k, $v);
+    }
+}
+
+function regen_xoStats()
+{
+    global $xoopsUser, $xoopsModule;
+
+    $stats = array();
+
+    /** @var \XoopsMemberHandler $memberHandler */
+    $memberHandler = xoops_getHandler('member');
+
+    // Getting Total Online Users
+    /** @var \XoopsOnlineHandler $online_handler */
+    $online_handler = xoops_getHandler('online');
+    // set gc probabillity to 10% for now..
+    if (mt_rand(1, 100) < 11) {
+        $online_handler->gc(300);
+    }
+    if (is_object($xoopsUser)) {
+        $uid   = $xoopsUser->getVar('uid');
+        $uname = $xoopsUser->getVar('uname');
+    } else {
+        $uid   = 0;
+        $uname = '';
+    }
+
+    $requestIp = IPAddress::fromRequest()->asReadable();
+    $requestIp = (false === $requestIp) ? '0.0.0.0' : $requestIp;
+    if (is_object($xoopsModule)) {
+        $online_handler->write($uid, $uname, time(), $xoopsModule->getVar('mid'), $requestIp);
+    } else {
+        $online_handler->write($uid, $uname, time(), 0, $requestIp);
+    }
+    $onlines = $online_handler->getAll();
+    if (empty($onlines)) {
+        $stats['totalonline'] = 0;
+    } else {
+        $totalonline   = count($onlines);
+        $stats['totalonline'] = $totalonline;
+    }
+
+    //Getting Total Registered Users
+    $level_criteria = new \Criteria('level', 0, '>');
+    $criteria = new \CriteriaCompo($level_criteria);
+    $criteria24 = new \CriteriaCompo($level_criteria);
+    $criteria48 = new \CriteriaCompo($level_criteria);
+    $totaluser = $memberHandler->getUserCount($level_criteria);
+    $stats['totaluser'] = $totaluser;
+
+    //Getting User Registration Statistics
+    $users_reg_24 = $memberHandler->getUserCount($criteria24->add(new \Criteria('user_regdate', (mktime(0, 0, 0) - (24 * 3600)), '>=')), 'AND');
+    $users_reg_48 = $memberHandler->getUserCount($criteria48->add(new \Criteria('user_regdate', (mktime(0, 0, 0) - (48 * 3600)), '>=')), 'AND');
+    $todayregister       = $users_reg_24;
+    $yesterdayregister   = $users_reg_48;
+    $stats['totalregisteredtoday'] = $todayregister;
+    $stats['totalregisteredyesterday'] = $yesterdayregister;
+
+    // Getting Last Registered Member
+    $limit        = 1;
+    $criteria->setOrder('DESC');
+    $criteria->setSort('user_regdate');
+    $criteria->setLimit($limit);
+    $lastmembers = $memberHandler->getUsers($criteria);
+    $lastusername = $lastmembers[0]->getVar('uname');
+    $lastrealname = $lastmembers[0]->getVar('name');
+    $latestuid = $lastmembers[0]->getVar('uid');
+    $latestmembername = $lastrealname;
+    $latestmemberuname = $lastusername;
```

Comment:
I'm not sure what's the difference between "$latestmemberuname" and "$lastusername". 

Are both names required somewhere in the already existing templates? Otherwise why don't we use one or the other, but consistently, i.e. everything starts with "last" or "latest"?

Reply:
There were a lot of extra assignments. Should be better now.